app/models/atmosphere/virtual_machine_template.rb

```diff
@@ -144,6 +144,12 @@ def perform_delete_in_cloud
       logger.warn("VMT with #{id_at_site} does not exist - continuing")
     end
 
+    # Returns the hourly cost for this template assuming a given VM flavor
+    def get_hourly_cost_for(flavor)
+      incarnation = flavor.flavor_os_families.find_by(os_family: appliance_type.os_family)
+      incarnation && incarnation.hourly_cost
+    end
```

Comment:
@mkasztelnik  @Nuanda I'm fine with fixing this problem right here and now but you need to tell me what you want me to do. Generally speaking, the situation we're dealing with here is a "can't happen" error, i.e. I do not expect `f.get_hourly_cost_for` to ever be invoked on an `OSFamily` object which is not already bound to `f`. This whole "guard against nils" code is just my paranoia - I want to be a good boy scout and not blindly invoke a custom method on an object which is passed to my code - simply because I do not control the requestor and cannot guarantee that the input argument is not blank. As already suggested, I can either respond with nil ("garbage in, garbage out") or toss an exception - whichever you prefer.